opentelemetry_api/opentelemetry/context/traceparent.h

```diff
@@ -0,0 +1,51 @@
+/* Copyright 2019, OpenTelemetry Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+#ifndef OPENTELEMETRY_CONTEXT_TRACEPARENT_H_
+#define OPENTELEMETRY_CONTEXT_TRACEPARENT_H_
+
+#include "../status.h"
+
+#ifdef __cplusplus
+namespace opentelemetry {
+namespace context {
+extern "C" {
+#endif
+
+#pragma pack(push, 1)
+typedef struct
+{
+  uint8_t version;
+  uint8_t trace_id[16];
+  uint8_t parent_id[8];
+  uint8_t trace_flags;
+} TraceParent;
```

Comment:
I think this should be a `class SpanContext` and there should be a helper function like:
```
SpanContext FromTraceParentHeader(string_view header);
```